openedx/core/djangoapps/certificates/api.py

```diff
@@ -0,0 +1,43 @@
+"""
+The public API for certificates.
+"""
+
+from openedx.core.djangoapps.certificates.config import waffle
+
+
+SWITCHES = waffle.waffle()
+
+
+def auto_certificate_generation_enabled():
+    return (
+        SWITCHES.is_enabled(waffle.SELF_PACED_ONLY) or
+        SWITCHES.is_enabled(waffle.INSTRUCTOR_PACED_ONLY)
+    )
+
+
+def auto_certificate_generation_enabled_for_course(course):
+    if not auto_certificate_generation_enabled():
+        return False
+
+    if course.self_paced:
```

Comment:
Is it still necessary at this point to distinguish between these 2 waffle switches?

Reply:
I think it makes sense to add the "unified" waffle switch in one PR, activate it on production, then change the code to actually use it in a subsequent PR so that we don't have any time during which auto-generated certs is turned back off, even for a couple minutes. I'd be in favor of adding that switch soon, either in this PR or one right after it.

Reply:
Yeah, I don't know.  Effectively, no, but I view this PR as just an intermediary between the current state and the point at which we reduce things down to a single waffle switch for auto-certs.

Reply:
That's a good point, Sandy.  We can add the switch in now and change the code later.

Reply:
You should be able to activate the new unified switch in stage/production even right now.  No reason for any code to be released before doing so.  This way, that switch will already be present, active and available for use - whenever you decide to deploy the code to start checking for it.

Reply:
Good point. I forgot that waffle switches can be activated without any code to declare their existence.